db/migrate/20161118150205_add_user_email_fingerprint.rb

```diff
@@ -0,0 +1,40 @@
+class AddUserEmailFingerprint < ActiveRecord::Migration
+  class User < ActiveRecord::Base
+  end
+
+  def up
+    add_column :users, :email_fingerprint, :string, default: ''
+    add_column :users, :encrypted_email, :text, default: ''
```

Comment:
We can discuss in the follow-up PR....my understanding is that if we change the cost, then we need to be able to update all the rows in our table to the new way. With a large enough database, this is not an instantaneous process (especially since these fingerprints are intended to be slow-ish) so we'd need to add a second column and then do a rename/swap or something and sync it with the deploy/change of the cost?

Anyways, not gonna block this PR, let's definitely discuss a follow-up PR but my understanding of how we'd want to handle this smoothly, in production, at a reasonable scale mean that I think we'd be best of storing cost per-row.